### PR TITLE
FRW-450 Auth documentation requirements

### PR DESCRIPTION
**Pull request by dimitriyTsemma**
## PR Description

https://spryker.atlassian.net/browse/FRW-450

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+### 2) Set up behavior
+
+Activate the following plugins:
+
+**Storefront API plugins**
+
+| PLUGIN | SPECIFICATION                                                                        | NAMESPACE                                                                                            |
+| --- |--------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| IsProtectedTableColumnExpanderPlugin | Provides capability to extends route description table.                              | Spryker\\Glue\\GlueStorefrontApiApplicationAuthorizationConnector\\Plugin\\GlueApplication           |
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:80]: It's a description that is valid for the plugin interface, not the plugin. Please adjust to a more specific one. (Applicable to BackendApi as well)

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+| AuthorizationContextExpanderPlugin | Adds the information about protected paths to the documentation generation context.  | Spryker\\Glue\\GlueStorefrontApiApplicationAuthorizationConnector\\Plugin\\DocumentationGeneratorApi |
+| ProtectedPathAuthorizationStrategyPlugin | Authorization rule for the route that uses the current strategy.                     | Spryker\\Client\\GlueStorefrontApiApplicationAuthorizationConnector\\Plugin\\Authorization                                         |
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:82]: Please update the plugin description to be more specific.

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+```php
+<?php
+
+namespace Pyz\Glue\GlueApplication;
+
+use Spryker\Glue\GlueBackendApiApplicationAuthorizationConnector\Plugin\GlueApplication\IsProtectedTableColumnExpanderPlugin as BackendIsProtectedTableColumnExpanderPlugin;
+use Spryker\Glue\GlueStorefrontApiApplicationAuthorizationConnector\Plugin\GlueApplication\IsProtectedTableColumnExpanderPlugin as StorefrontIsProtectedTableColumnExpanderPlugin;
+
+class GlueApplicationDependencyProvider extends SprykerGlueApplicationDependencyProvider
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:189]: `use` is missing for `SprykerGlueApplicationDependencyProvider`

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+```php
+<?php
+
+namespace Pyz\Glue\DocumentationGeneratorApi;
+
+use Spryker\Glue\GlueBackendApiApplicationAuthorizationConnector\Plugin\DocumentationGeneratorApi\AuthorizationContextExpanderPlugin as BackendAuthorizationContextExpanderPlugin;
+use Spryker\Glue\GlueStorefrontApiApplicationAuthorizationConnector\Plugin\DocumentationGeneratorApi\AuthorizationContextExpanderPlugin as StorefrontAuthorizationContextExpanderPlugin;
+
+class DocumentationGeneratorApiDependencyProvider extends SprykerDocumentationGeneratorApiDependencyProvider
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:148]: `use` is missing for `SprykerDocumentationGeneratorApiDependencyProvider`

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+     * @var string
+     */
+    protected const GLUE_STOREFRONT_API_APPLICATION_NAME = 'storefront';
+    
+    /**
+     * @param \Spryker\Glue\DocumentationGeneratorApi\Expander\ContextExpanderCollectionInterface $contextExpanderCollection
+     *
+     * @return \Spryker\Glue\DocumentationGeneratorApi\Expander\ContextExpanderCollectionInterface
+     */
+    protected function getContextExpanderPlugins(ContextExpanderCollectionInterface $contextExpanderCollection): ContextExpanderCollectionInterface
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:165]: `use` is missing for `ContextExpanderCollectionInterface`

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+description: This document describes how to use a Protected endpoints authorization strategy for Storefront API application and Backend API application into a Spryker project.
+last_updated: October 21, 2022
+template: feature-integration-guide-template
+---
+
+This document describes how to use a Protected endpoints authorization strategy for Storefront API application and Backend API application into a Spryker project.
+
+## Install feature core
+
+Follow the steps below to install the Authorization core.
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:12]: > Authorization core

It's not the name of the feature that we are integrating in this document

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+**src/Pyz/Zed/Authorization/****AuthorizationDependencyProvider****.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\Authorization;
+
+use Spryker\Zed\GlueBackendApiApplicationAuthorizationConnector\Communication\Plugin\Authorization\ProtectedPathAuthorizationStrategyPlugin;
+
+class AuthorizationDependencyProvider extends SprykerAuthorizationDependencyProvider
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:124]: `use` for `SprykerAuthorizationDependencyProvider` is missing.

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+**src/Pyz/Client/Authorization/****AuthorizationDependencyProvider****.php**
+
+```php
+<?php
+
+namespace Pyz\Client\Authorization;
+
+use Spryker\Client\GlueStorefrontApiApplicationAuthorizationConnector\Plugin\Authorization\ProtectedPathAuthorizationStrategyPlugin;
+
+class AuthorizationDependencyProvider extends SprykerAuthorizationDependencyProvider
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:101]: `use` for `SprykerAuthorizationDependencyProvider` is missing.

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
+---
+title: Security and Authentication
+description: This article describes the authorization mechanism used in Spryker, as well as user scopes, database tables, and extension points.
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:3]: > database tables, and extension points

Do we really have it in the document?

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+last_updated: Oct 24, 2022
+template: glue-api-storefront-guide-template
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:5]: Where do we have this template?

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
+---
+title: Authentication and authorization
+description: Information about Glue API authentication and authorization.
+last_updated: Oct 23, 2022
+template: glue-api-storefront-guide-template
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md:5]: [Informational] We have both `storefront` and `backend` API in one document, which is located in `backend-api` section. This should be refactored in the following PR's. CC @gechetspr

**Review by asmarovydlo — CHANGES_REQUESTED** [review]

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md (hunk 11 above)
Comment (dimitriyTsemma) [reply docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md:5]: Yes, we discussed it with @gechetspr

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md (hunk 10 above)
Comment (dimitriyTsemma) [reply docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:5]: _templates/glue-api-storefront-guide-template.md

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+    protected function getAuthorizationStrategyPlugins(): array
+    {
+        return [
+            new ProtectedPathAuthorizationStrategyPlugin(),
+        ];
+    }
+}
+```
+
+**src/Pyz/Glue/DocumentationGeneratorApi/****DocumentationGeneratorApiDependencyProvider****.php**
```
Comment (gechetspr) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:138]: TBH documentation is not required for auth process. So we can not include it here

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+The default lifetime of the access tokens is 8 hours (28800 seconds) and 1 month (2628000 seconds) of the refresh tokens.
+
+For security purposes, when you finish sending requests as a user, or if a token gets compromised, we recommend revoking the refresh token. Revoked tokens are marked as expired on the date and time of the request and can no longer be exchanged for access tokens.
+
+Expired tokens are stored in the database, and you can configure them to be deleted.
+
+## Authentication in the Storefront and Backend applications
+
+Below, you can find information about [Backend and Storefront API module differences](/docs/scos/dev/glue-api-guides/{{page.version}}/glue-backend-api/how-to-guides/backend-and-storefront-api-module-differences.html) and [Authentication integration.](/docs/scos/dev/feature-integration-guides/{{page.version}}/glue-api/glue-backend-api/glue-api-authentication-integration.html)
```
Comment (gechetspr) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md:26]: Do we have this doc already (backend-and-storefront-api-module-differences.html)?

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+The authorization mechanism provided by the Glue is flexible and allows you to define which endpoints specifically require authentication. Usually, protected endpoints include customer wish lists, carts, personal data and the like. Also, you may want to protect additional areas as required by your project. In addition to endpoints, you may require authorization to use specific REST verbs. For example, a certain endpoint can allow retrieving information, but not modifying it. In this case, the GET verb can be allowed for usage without authentication, while the PUT, PATCH and DELETE verbs will require user authentication to use.
+
+## How authentication works
+
+To authenticate a user, the client must send an authentication request, containing the username and password. If the provided credentials match a valid Spryker user, the API responds with a 201 response code and a message containing an access token, also known as a bearer token. The token can then be used to access protected API resources. Schematically, the authentication and authorization scheme of Spryker REST API can be represented as follows:
```
Comment (gechetspr) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:14]: not completely true. It depends on grant type. Username and password are required only for the `password` grant type

**Review by gechetspr — CHANGES_REQUESTED** [review]

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md (hunk 13 above)
Comment (dimitriyTsemma) [reply docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md:26]: yes

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md (hunk 12 above)
Comment (dimitriyTsemma) [reply docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:138]: just delete this one and 2 related plugins AuthorizationContextExpanderPlugin ?

Comment (gechetspr) [reply docs/scos/dev/feature-integration-guides/202204.0/glue-api/glue-backend-api/glue-api-authorization-protected-endpoints-integration.md:140]: Yes. Documentation is related for documentation how-to.

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md (hunk 13 above)
Comment (gechetspr) [reply docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/authentication-and-authorization.md:26]: Cool

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-authentication-server.md [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+
+```php
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Spryker\Zed\AuthenticationOauth\Communication\Plugin;
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-authentication-server.md:29]:
```suggestion
namespace Spryker\Zed\AuthenticationOauth\Communication\Plugin\Authentication;
```

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-authentication-server.md [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+         return $this->getFacade()->authenticate($glueAuthenticationRequestTransfer);
+    }
+}
+
+```
+</details>
+
+This plugin is run by `AuthenticationFacade::authenticate()`. 
+
+For the Glue Storefront API application `Spryker\Client\AuthenticationOauth\Communication\Plugin\OauthAuthenticationServerPlugin` is used and run by `AuthenticationClient::authenticate()`.
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-authentication-server.md:86]:
```suggestion
For the Glue Storefront API application `Spryker\Client\AuthenticationOauth\Communication\Plugin\Authentication\OauthAuthenticationServerPlugin` is used and run by `AuthenticationClient::authenticate()`.
```

**Review by asmarovydlo — APPROVED** [review]

**Review by gechetspr — APPROVED** [review]

**Review by asmarovydlo — APPROVED** [review]

**Review by VadymSachenko — CHANGES_REQUESTED** [review]

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+
+![auth-scheme.png](https://spryker.s3.eu-central-1.amazonaws.com/docs/Glue+API/Glue+API+Developer+Guides/Security+and+Authentication/auth-scheme.png)
+
+Access tokens issued by the API have a limited lifetime. When a token is issued, the response message contains not only the access token but also the duration of its validity, in seconds, and a refresh token. Once the access token expires, the corresponding refresh token can be exchanged for a new pair of access and refresh tokens.
+
+If an invalid or expired token is passed when accessing a protected resource, or no token is passed at all, the API responds with the `401 Unauthorized response` code. This response code also is issued if a user is not authorized to access that particular resource. The response body contains a detailed error message. It is, therefore, the client's responsibility to handle the `401` response code and error messages correctly.
```
Comment (VadymSachenko) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:20]:
```suggestion
If an invalid or expired token is passed when accessing a protected resource, or no token is passed at all, the API responds with the `401 Unauthorized` response code. This response code also is issued if a user is not authorized to access that particular resource. The response body contains a detailed error message. It is, therefore, the client's responsibility to handle the `401` response code and error messages correctly.
```

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-an-authentication-server.md [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+```
+</details>
+
+This plugin is run by `AuthenticationFacade::authenticate()`. 
+
+For the Glue Storefront API application `Spryker\Client\AuthenticationOauth\Communication\Plugin\Authentication\OauthAuthenticationServerPlugin` is used and run by `AuthenticationClient::authenticate()`.
+
+Connect it the following way in order to enable provided authentication server for the Glue Backend API application:
```
Comment (VadymSachenko) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-an-authentication-server.md:88]: Connect what? What do you mean by "it"?
```suggestion
To enable provided authentication server for the Glue Backend API application, connect it in the following way:
```

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-an-authentication-server.md [hunk 19]
```diff
@@ -0,0 +1,133 @@
+---
+title: How to use authentication server
```
Comment (VadymSachenko) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-an-authentication-server.md:2]:
```suggestion
title: How to use an authentication server
```

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-an-authentication-server.md [hunk 20]
```diff
@@ -0,0 +1,133 @@
...
+template: howto-guide-template
+redirect_from:
+  - /docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/using-authentication-server.html
+---
+
+This document describes how to use an authentication server.
+
+Integrate authentication following the [Glue API Authentication integration](/docs/scos/dev/technical-enhancement-integration-guides/glue-authentication-integration.html) guide.
+
+Glue allows switching between different authentication servers. In order to provide it the `AuthenticationServerPluginInterface` must be implemented.
```
Comment (VadymSachenko) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-use-an-authentication-server.md:14]: What do you mean by "it" in the second sentence?
```suggestion
Glue allows switching between different authentication servers. In order to provide it, the `AuthenticationServerPluginInterface` must be implemented.
```

**Review by VadymSachenko — CHANGES_REQUESTED** [review]

File: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md [hunk 21]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (VadymSachenko) [inline docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:1]: we have almost the same document here docs\scos\dev\glue-api-guides\202204.0\security-and-authentication.md. Which one should be removed? Or you want to let both live for some reason?

Comment (olhalivitchuk) [reply docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:1]: Seems like `docs\scos\dev\glue-api-guides\202204.0\security-and-authentication.md` is outdated.

@gechetspr what do you think?

Comment (gechetspr) [reply docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md:1]: We need them both for now. They are similar, but for the current and deprecated approach. We will remove old one, when the new infrastructure will be widely used

**Review by VadymSachenko — APPROVED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**